### PR TITLE
Wasm support: stub libc calls in `miniz_oxide_c_api`

### LINKED ISSUE #34: Wasm: stub the libc dependency in miniz_oxide_c_api
Hi!

I read through #26, and we could add a small hack to stub the left-over libc calls to also be able to compile `miniz_oxide_c_api` to Wasm. This is all following what was done to use libsodium in [alexcrichton/wasm-sodium](https://github.com/alexcrichton/wasm-sodium) (see the [libc-stub](https://github.com/alexcrichton/wasm-sodium/tree/master/libc-stub) folder).

The [wasm32 branch in wehlutyk/miniz_oxide](https://github.com/wehlutyk/miniz_oxide/tree/wasm32) shows that, and it makes compiling `flate2-rs` to wasm32 possible (see [wehlutyk/flate2-rs/tree/wasm32](https://github.com/wehlutyk/flate2-rs/tree/wasm32)). I use it to read compressed pdf files using `lopdf`, all in the browser, and up to now it works like a charm.

Would you accept a PR for this? It would be great as we could then merge the changes for `flate2-rs` and `lopdf` and have them work on wasm32. But seeing as it's a bit of hack I thought you might want to wait till libc gets implementations for wasm32. Then again it's a useful shim in the meantime :)

### PR DESCRIPTION
**Pull request by wehlutyk**
This should fix #34. I'm pretty new to the Rust community (I think it's my first Rust PR), so there might be some ironing out to do here. Feel free to point me to what needs doing :)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,58 @@
+#![feature(allocator_api)]
+
```
[the remark this answers is not in the document]
Comment (oyvindln) [reply src/lib.rs]: Good point.
Also, the feature should only be enabled on wasm32 ideally.

[the remark this answers is not in the document]
Comment (eminence) [reply src/lib.rs]: Or maybe only enabled if the libc-stub feature is in-use?

**Comment by alexcrichton** [comment]
ping @Frommi, would it be possible to publish this on crates.io?

**Comment by Frommi** [comment]
Hi @alexcrichton . Sorry for the silent treatment. Yeah, I will look into it tomorrow evening.

**Comment by alexcrichton** [comment]
No worries, and thanks!

**Pull request merged**